File: App/connection.py
import requests

from user import User

# === Login ===
def login(SERVER_IP):
    username = input("Username: ")
    password = input("Password: ")

    try:
        user = User(username, password)
        res = requests.post(f"http://{SERVER_IP}/login", json={
            "username": username,
            "password": password
        })

        if res.status_code == 200:
            data = res.json()
            print("Logged in successfully!")
            user.session_id = data['sessionId']
            return user
        else:
            print("Login failed:", res.json().get("message", "Unknown error"))
            del user
            return None
    except Exception as e:
        print("Error connecting to server:", e)
        return None

File: Server/connection.js
import http from 'http';
import express from 'express';
import bodyParser from 'body-parser';

import { users, sessions } from './db.js';

export const app = express();
export const server = http.createServer(app);
app.use(bodyParser.json());

export const PORT = 3000;

//#region LOGIN
app.post('/login', (req, res) => {
  const { username, password } = req.body;
  const user = users.find(u => u.username === username && u.password === password);
  if (user) {
    const sessionId = `${Date.now()}-${Math.random()}`;
    sessions[sessionId] = user;
    res.json({ success: true, sessionId });
  } else {
    res.status(401).json({ success: false, message: 'Invalid credentials' });
  }
});
//#endregion
